tests/Operation/FindTest.php

```diff
@@ -154,4 +156,62 @@ public function provideInvalidConstructorCursorTypeOptions()
     {
         return $this->wrapValuesForDataProvider([-1, 0, 4]);
     }
+
+    public function testExplainableCommandDocument(): void
+    {
+        // all options except deprecated "snapshot" and "maxScan"
+        $options = [
+            'allowDiskUse' => true,
+            'allowPartialResults' => true,
+            'batchSize' => 123,
+            'collation' => ['locale' => 'fr'],
+            'comment' => 'explain me',
+            'hint' => '_id_',
+            'limit' => 15,
+            'max' => ['x' => 100],
+            'maxTimeMS' => 100,
+            'min' => ['x' => 10],
+            'noCursorTimeout' => true,
+            'oplogReplay' => true,
+            'projection' => ['_id' => 0],
+            'readConcern' => new ReadConcern(ReadConcern::LOCAL),
+            'returnKey' => true,
+            'showRecordId' => true,
+            'skip' => 5,
+            'sort' => ['x' => 1],
+            'let' => ['y' => 2],
+            // Intentionally omitted options
+            'cursorType' => Find::NON_TAILABLE,
+            'maxAwaitTimeMS' => 500,
+            'modifiers' => ['foo' => 'bar'],
+            'readPreference' => new ReadPreference(ReadPreference::SECONDARY_PREFERRED),
+            'typeMap' => ['root' => 'array'],
+        ];
+        $operation = new Find($this->getDatabaseName(), $this->getCollectionName(), ['x' => 1], $options);
+
+        $expected = [
+            'find' => $this->getCollectionName(),
+            'filter' => (object) ['x' => 1],
+            'allowDiskUse' => true,
+            'allowPartialResults' => true,
+            'batchSize' => 123,
```

Comment:
The `batchSize` seems to be useless for an explain. It is not supported by the aggregate command with `explain` option:
https://github.com/mongodb/mongo-php-library/blob/78cc92542e8710925e6a8f8cffee54051f4385b9/src/Operation/Aggregate.php#L225-L229

Reply:
> The batchSize seems to be useless for an explain

I imagine so, but I don't think there's any reason for PHPLIB to care about that. We can rely on the server to ignore it or raise an error. I wouldn't expect an error (not going to bother testing myself), but if so the user could always revise their operation options.

The case for unsetting `batchSize` when using the `explain` option with `aggregate` is different. That logic has nothing to do with it being relevant/supported when explaining a pipeline, but rather the fact that `aggregate` with `explain: true` returns a single document. The `batchSize` option is only relevant if `aggregate` yields a cursor.